libbpf-rs/src/query.rs

```diff
@@ -122,36 +122,60 @@ impl From<&libbpf_sys::bpf_line_info> for LineInfo {
 #[repr(C)]
 pub struct Tag(pub [u8; 8]);
 
-/// Information about a BPF program
+/// Information about a BPF program. Maps to `struct bpf_prog_info` in kernel uapi.
 #[derive(Debug, Clone)]
-// TODO: Document members.
-#[allow(missing_docs)]
 pub struct ProgramInfo {
+    /// A user-defined name for the BPF program (null-terminated string).
     pub name: CString,
+    /// The type of the program.
     pub ty: ProgramType,
+    /// An 8-byte hash (`BPF_TAG_SIZE`) computed from the program’s
+    /// contents; used to detect changes in the program code.
     pub tag: Tag,
+    /// A unique identifier for the program instance.
     pub id: u32,
+    /// Pointer to the JIT-compiled instructions
     pub jited_prog_insns: Vec<u8>,
+    /// Pointer to the translated BPF instructions in an intermediate
     pub xlated_prog_insns: Vec<u8>,
-    /// Duration since system boot
+    /// Time (since system boot) at which the program was loaded.
     pub load_time: Duration,
+    /// UID of the user who loaded the program.
     pub created_by_uid: u32,
+    /// Pointer to an array of map IDs associated with this program.
     pub map_ids: Vec<u32>,
+    /// Network interface index if the program is attached to a specific device.
     pub ifindex: u32,
+    /// Whether the program is gpl compatible.
     pub gpl_compatible: bool,
+    /// Device ID of the network namespace that the program is associated with.
     pub netns_dev: u64,
+    /// Inode number of the network namespace associated with the program.
     pub netns_ino: u64,
+    /// Number of kernel symbols in the JITed code (if available).
     pub jited_ksyms: Vec<*const c_void>,
+    /// Number of function length records available for the JITed code.
     pub jited_func_lens: Vec<u32>,
+    /// Identifier of the associated BTF (BPF Type Format) data.
     pub btf_id: u32,
+    /// Size (in bytes) of each record in the function info array.
     pub func_info_rec_size: u32,
+    /// Pointer to an array of function info records for this program.
     pub func_info: Vec<libbpf_sys::bpf_func_info>,
+    /// Pointer to an array of line info records mapping BPF instructions
+    /// to source code lines.
     pub line_info: Vec<LineInfo>,
+    /// Pointer to line info records for the JIT-compiled code.
     pub jited_line_info: Vec<*const c_void>,
+    /// Size (in bytes) of each line info record.
     pub line_info_rec_size: u32,
+    /// Size (in bytes) of each record in the JITed line info array.
     pub jited_line_info_rec_size: u32,
+    /// Pointer to an array of program tags.
     pub prog_tags: Vec<Tag>,
+    /// Total accumulated run time (in nanoseconds) for the program’s execution.
     pub run_time_ns: u64,
```

Comment:
Well, looking at it this should probably be a `Duration`. But of course you didn't change the type so it's fine to keep for now. Just pointing it out in case you want to fix it up subsequently.